apis/Google.Cloud.Firestore/Google.Cloud.Firestore.Tests/AggregateQueryTest.cs

```diff
@@ -0,0 +1,68 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License").
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at 
+//
+// https://www.apache.org/licenses/LICENSE-2.0 
+//
+// Unless required by applicable law or agreed to in writing, software 
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and 
+// limitations under the License.
+
+using Google.Api.Gax.Grpc;
+using Google.Cloud.ClientTesting;
+using Google.Cloud.Firestore.V1;
+using Moq;
+using System.Threading.Tasks;
+using Xunit;
+using static Google.Cloud.Firestore.Tests.ProtoHelpers;
+using static Google.Cloud.Firestore.V1.StructuredAggregationQuery.Types;
+using static Google.Cloud.Firestore.V1.StructuredAggregationQuery.Types.Aggregation.Types;
+
+namespace Google.Cloud.Firestore.Tests;
+public class AggregateQueryTest
+{
+    private static readonly FirestoreDb s_db = FirestoreDb.Create("proj", "db", new FakeFirestoreClient());
+    private static readonly Query s_mockQuery = s_db.Collection("col");
+
+    [Fact]
+    public void TestToStructuredAggregationQuery()
+    {
+        var expectedStructuredAggregationQuery = new StructuredAggregationQuery
+        {
+            StructuredQuery = s_mockQuery.ToStructuredQuery(),
+            Aggregations = { new Aggregation { Alias = "Count", Count = new Count() } }
+        };
+        Assert.Equal(expectedStructuredAggregationQuery, s_mockQuery.Count().ToStructuredAggregationQuery());
+    }
+
+    [Fact]
+    public async Task GetSnapshotAsync_VerifySnapshotMembers()
+    {
+        Mock<FirestoreClient> mock = new() { CallBase = true };
+        var sampleReadTime = CreateProtoTimestamp(1, 3);
+        var response = new FakeAggregationQueryStream(new[]
+        {
+            new RunAggregationQueryResponse { ReadTime = sampleReadTime, Result = new AggregationResult() }
+        });
+        mock.Setup(c => c.RunAggregationQuery(It.IsAny<RunAggregationQueryRequest>(), It.IsAny<CallSettings>())).Returns(response);
+
+        var db = FirestoreDb.Create("proj", "db", mock.Object);
+        var query = db.Collection("col").Select("Name");
+        var expectedQuery = query.Count();
+        var snapshot = await expectedQuery.GetSnapshotAsync();
+        Assert.Equal(expectedQuery, snapshot.Query);
+        Assert.Equal(Timestamp.FromProto(sampleReadTime), snapshot.ReadTime);
+    }
+
+    [Fact]
+    public void Equality()
+    {
+        var aggQueryWithCount = s_mockQuery.Count();
+        var aggQueryWithoutCount = new AggregateQuery(s_mockQuery);
+        EqualityTester.AssertEqual(aggQueryWithCount, equal: new[] { new AggregateQuery(s_mockQuery).WithAggregation(new Aggregation { Count = new Count() }) }, unequal: new[] { new AggregateQuery(s_db.Collection("col1")), aggQueryWithoutCount });
```

Comment:
Could you please elaborate a bit more on this.

Reply:
Which part? The first is just "use some separate local variables for clarity instead of doing everything in a single huge statement".
The second is about creating entirely independent but equal queries: `s_db.Collection("xyz")` and calling `s_db.Collection("xyz")` again, so equal underlying queries rather than a single Query object used in both cases.

Reply:
Got it Thanks